vendor/github.com/openshift/library-go/pkg/proc/proc_linux.go

```diff
@@ -0,0 +1,82 @@
+package proc
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// parseProcForZombies parses the current procfs mounted at /proc
+// to find processes in the zombie state.
+func parseProcForZombies() ([]int, error) {
+	files, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return nil, err
+	}
+
+	var zombies []int
+	for _, file := range files {
+		processID, err := strconv.Atoi(file.Name())
+		if err != nil {
+			break
```

Comment:
How can we be sure there are no file names in procfs that get sorted before numerics?

Reply:
Seems like there is no guarantee since `ioutil.ReadDir(...)` will [sort by name ](https://golang.org/src/io/ioutil/ioutil.go?s=3208:3259#L90)(letters before numbers). The `Atoi` err check would break out of the loop on any non-numerical named procfs files. I suppose the `files` list could be better sorted first, or the non-numerical filenames could be removed from the list. @Miciah do you bring this up out of time complexity concerns, or just trying to improve the vendored code?

Reply:
I'm wondering whether the vendored code is correct.  For example, if /proc had a file named `.foo`, it would be sorted before the numeric names, and the loop would break before scanning any pids.

Reply:
@mrunalp, am I missing something?  Does some API guarantee exist for procfs that prohibits file names that would be sorted before numeric names?